File: frontend/src/types/ollama.ts
export interface OllamaModel {
  name: string
  size?: number
  digest?: string
  modified_at?: string
}

export interface OllamaModelListResponse {
  models: OllamaModel[]
}

export interface OllamaStatusResponse {
  connected: boolean
  url: string
  models_count?: number
  error?: string
}

File: backend/app/schemas/ollama.py
"""
Pydantic schemas for Ollama API.
"""
from typing import List, Optional

from pydantic import BaseModel, Field


class OllamaModel(BaseModel):
    """Schema for an Ollama model."""

    name: str
    size: Optional[int] = None
    digest: Optional[str] = None
    modified_at: Optional[str] = None


class OllamaModelListResponse(BaseModel):
    """Schema for Ollama model list response."""

    models: List[OllamaModel]


class OllamaStatusResponse(BaseModel):
    """Schema for Ollama connection status."""

    connected: bool
    url: str
    models_count: Optional[int] = None
    error: Optional[str] = None
